src/libraries/Microsoft.Extensions.Logging/gen/LoggerMessageGenerator.Emitter.cs

```diff
@@ -0,0 +1,591 @@
+                            1 => "global::Microsoft.Extensions.Logging.LogLevel.Debug",
+                            2 => "global::Microsoft.Extensions.Logging.LogLevel.Information",
+                            3 => "global::Microsoft.Extensions.Logging.LogLevel.Warning",
+                            4 => "global::Microsoft.Extensions.Logging.LogLevel.Error",
+                            5 => "global::Microsoft.Extensions.Logging.LogLevel.Critical",
+                            6 => "global::Microsoft.Extensions.Logging.LogLevel.None",
+                            _ => $"(global::Microsoft.Extensions.Logging.LogLevel){lm.Level}",
+                        };
+                    }
+
+                    return level;
+                }
+            }
+
+            private void GenEnumerationHelper(LoggerClass lc)
+            {
+                foreach (LoggerMethod lm in lc.Methods)
+                {
+                    if (UseLoggerMessageDefine(lm))
+                    {
+                        foreach (LoggerParameter p in lm.TemplateParameters)
+                        {
+                            if (p.IsEnumerable)
+                            {
+                                _builder.Append($@"
+        [{_generatedCodeAttribute}]
+        private static string __Enumerate(global::System.Collections.IEnumerable? enumerable)
+        {{
+            if (enumerable == null)
+            {{
+                return ""(null)"";
+            }}
+
+            var sb = new global::System.Text.StringBuilder();
+            _ = sb.Append('[');
+
+            bool first = true;
+            foreach (object e in enumerable)
+            {{
+                if (!first)
+                {{
+                    _ = sb.Append("", "");
+                }}
+
+                if (e == null)
+                {{
+                    _ = sb.Append(""(null)"");
+                }}
+                else
+                {{
+                    if (e is global::System.IFormattable fmt)
+                    {{
+                        _ = sb.Append(fmt.ToString(null, global::System.Globalization.CultureInfo.InvariantCulture));
+                    }}
+                    else
+                    {{
+                        _ = sb.Append(e);
+                    }}
+                }}
+
+                first = false;
+            }}
+
+            _ = sb.Append(']');
+
+            return sb.ToString();
+        }}
+");
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    internal static class StringHelper
+    {
+        // Replace operations need to be done with StringComparison.Ordinal
+        // (Adding code snippet below, since the API taking StringComparison is not available in netstandard2.0)
+        internal static string ReplaceOrdinal(this string s, string oldValue, string newValue)
```

Comment:
Why is this method here, but `ConvertEndOfLineAndQutationCharactersToEscapeForm` is static in the emitter class?

Also, who is calling this method?

Reply:
Forgot to remove this after adding **ConvertEndOfLineAndQutationCharactersToEscapeForm**